### PR TITLE
Improve README and overall documentation

### LINKED ISSUE #7: Improve the README
▪︎ Installation
▪︎ Example or link to example DAGs
▪︎ Link to the changelog
▪︎ License

### PR DESCRIPTION
**Pull request by tatiana**
Closes #7

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,50 +1,16 @@
...
-.. list-table::
-   :header-rows: 1
+* **DatabricksWorkflowTaskGroup**: Airflow `Task Group <https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#taskgroups>`_ that allows users to create a `Databricks Workflow <https://www.databricks.com/product/workflows>`_.
+* **DatabricksNotebookOperator**: Airflow `Operator <https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/operators.html>`_ which abstracts a pre-existing `Databricks Notebook <https://docs.databricks.com/notebooks/>`_. Can be used independently to run the Notebook, or within a Databricks Workflow Task Group.
+* **AstroDatabricksPlugin**: An Airflow `Plugin <https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html>`_ which is installed by the default. It allows users, by using the UI, to view a Databricks job and retry running it in case of failure.
 
-   * - Operator/Sensor Class
-     - Import Path
-     - Example DAG
+Known limitations
```
Comment (tatiana) [inline CHANGELOG.rst:13]: Need to add a note about `XCom`

File: README.md [hunk 2]
```diff
@@ -1,5 +1,106 @@
...
+   * `Shared/Notebook_2`
+
+2. Generate a [Databricks Personal Token](https://docs.databricks.com/dev-tools/auth.html#databricks-personal-access-tokens). This will be used in step (6). 
+
+3. Ensure that your Airflow environment is set up correctly by running the following commands:
+
+    ```shell
+    export AIRFLOW_HOME=`pwd`
+    export AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES": "airflow\\.* astro_databricks\\.*"
+
```
Comment (tatiana) [inline README.md:47]: Add note about pickling and Airflow versions under 2.5

File: README.md [hunk 3]
```diff
@@ -1,5 +1,106 @@
...
+    }'
+```
+
+5. Copy the following workflow into a file named `example_databricks_workflow.py` and add it to the `dags` directory of your Airflow project:
+   
+   https://github.com/astronomer/astro-providers-databricks/blob/45897543a5e34d446c84b3fbc4f6f7a3ed16cdf7/example_dags/example_databricks_workflow.py#L48-L101
+
+   Alternatively, you can download `example_databricks_workflow.py`
+   ```shell
+    curl -O https://raw.githubusercontent.com/astronomer/astro-providers-databricks/main/example_dags/example_databricks_workflow.py
```
Comment (tatiana) [inline README.md:69]: Update to the new `example_dags` directory

**Review by pankajkoti — APPROVED** [review]
Looks good to me. Would be nice to also add the note regarding enabling xcom pickling for airflow users <2.5 . Looks great, thank you!

File: README.md [hunk 4]
```diff
@@ -1,5 +1,106 @@
...
 
-For up-to-date documentation, visit https://astronomer.github.io/astro-providers-databricks/
\ No newline at end of file
+
+**Astro Databricks** is an [Apache Airflow](https://github.com/apache/airflow) provider created by [Astronomer](https://www.astronomer.io/) for an **optimal Databricks experience**.  With the `DatabricksTaskGroup`, Astro Datatricks allows you to run from Databricks workflows without
+the need of running Jobs individually, which can result in **75% cost reduction**.
+
+## Prerequisites
+
+* Apache Airflow >= 2.2
```
Comment (pankajkoti) [inline README.md:23]:
```suggestion
* Apache Airflow >= 2.2.4
```

File: src/astro_databricks/operators/__init__.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from astro_databricks.operators.notebook import DatabricksNotebookOperator
+from astro_databricks.operators.workflow import DatabricksWorkflowTaskGroup
+
+__all__ = [
+    "DatabricksNotebookOperator",
```
Comment (tatiana) [inline src/astro_databricks/operators/__init__.py:5]: Update example DAGs to use this

File: README.md (hunk 2 above)
Comment (tatiana) [reply README.md:47]: No longer necessary 🎉 
https://github.com/astronomer/astro-providers-databricks/pull/15

File: README.md (hunk 3 above)
Comment (tatiana) [reply README.md:69]: We'll rever the `example_dags` change to move to the root directory, so it won't be necessary

**Review by dimberman — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1,5 +1,106 @@
...
+   * `Shared/Notebook_1`
+   * `Shared/Notebook_2`
+
+2. Generate a [Databricks Personal Token](https://docs.databricks.com/dev-tools/auth.html#databricks-personal-access-tokens). This will be used in step (6). 
+
+3. Ensure that your Airflow environment is set up correctly by running the following commands:
+
+    ```shell
+    export AIRFLOW_HOME=`pwd`
+    export AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES": "airflow\\.* astro_databricks\\.*"
```
[the remark this answers is not in the document]
Comment (tatiana) [reply README.md]: I had the same thought, @dimberman , just logged a ticket so we don't lose track: https://github.com/astronomer/astro-providers-databricks/issues/16

File: pyproject.toml [hunk 7]
```diff
@@ -58,8 +57,8 @@ tests = [
 
 # TODO: change the name of the package
 [project.urls]
-Homepage = "https://github.com/astronomer/astronomer-sdk-template"
-Documentation = "https://github.com/astronomer/astronomer-sdk-template"
+Homepage = "https://github.com/astronomer/astro-providers-databricks/"
```
[the remark this answers is not in the document]
Comment (tatiana) [reply pyproject.toml:66]: Thanks, @dimberman , I've brought this discussion to the slack team channel - we can make this change in a separate PR once it is confirmed.

**Pull request merged**